From df744cf5b616fe3014a507af4669e93c3fd0b45b Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Wed, 28 Oct 2026 13:49:07 +0000
Subject: [PATCH 274/277] addons: accept URLs as add-on assets

An add-on's icon, fanart, screenshots and other art in <assets> are
paths inside the add-on or, from a repository, inside its art
directory. A web app's art is the service's own logo, which its owner
publishes over HTTPS and nobody else may redistribute. An asset given
as an http or https URL is now kept as it is, and the texture cache
fetches it like any remote thumbnail.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 xbmc/addons/addoninfo/AddonInfoBuilder.cpp | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

--- a/xbmc/addons/addoninfo/AddonInfoBuilder.cpp
+++ b/xbmc/addons/addoninfo/AddonInfoBuilder.cpp
@@ -45,6 +45,13 @@ std::string GetSharedLibraryNameRegexPat
   // linux is different and has the version number after the suffix
   return "^.*" + suffix + R"(\.?\d*\.?\d*\.?\d*$)";
 }
+
+std::string GetAssetPath(const std::string& basePath, const std::string& asset)
+{
+  if (URIUtils::IsHTTP(asset))
+    return asset;
+  return URIUtils::AddFileToFolder(basePath, asset);
+}
 }
 
 namespace ADDON
@@ -451,33 +458,33 @@ bool CAddonInfoBuilder::ParseXML(const A
           if (value == "icon")
           {
             if (elementsAssets->GetText() != nullptr)
-              addon->m_icon = URIUtils::AddFileToFolder(assetBasePath, elementsAssets->GetText());
+              addon->m_icon = GetAssetPath(assetBasePath, elementsAssets->GetText());
           }
           else if (value == "screenshot")
           {
             if (elementsAssets->GetText() != nullptr)
-              addon->m_screenshots.emplace_back(URIUtils::AddFileToFolder(assetBasePath, elementsAssets->GetText()));
+              addon->m_screenshots.emplace_back(
+                  GetAssetPath(assetBasePath, elementsAssets->GetText()));
           }
           else if (value == "fanart")
           {
             if (elementsAssets->GetText() != nullptr)
-              addon->m_art[value] = URIUtils::AddFileToFolder(assetBasePath, elementsAssets->GetText());
+              addon->m_art[value] = GetAssetPath(assetBasePath, elementsAssets->GetText());
           }
           else if (value == "banner")
           {
             if (elementsAssets->GetText() != nullptr)
-              addon->m_art[value] = URIUtils::AddFileToFolder(assetBasePath, elementsAssets->GetText());
+              addon->m_art[value] = GetAssetPath(assetBasePath, elementsAssets->GetText());
           }
           else if (value == "clearlogo")
           {
             if (elementsAssets->GetText() != nullptr)
-              addon->m_art[value] = URIUtils::AddFileToFolder(assetBasePath, elementsAssets->GetText());
+              addon->m_art[value] = GetAssetPath(assetBasePath, elementsAssets->GetText());
           }
           else if (value == "thumb")
           {
             if (elementsAssets->GetText() != nullptr)
-              addon->m_art[value] =
-                  URIUtils::AddFileToFolder(assetBasePath, elementsAssets->GetText());
+              addon->m_art[value] = GetAssetPath(assetBasePath, elementsAssets->GetText());
           }
         }
       }
